aequilibrae/paths/traffic_assignment.py

```diff
@@ -285,7 +303,7 @@ def set_cores(self, cores: int) -> None:
             cores (:obj:`int`): Number of CPU cores to use
         """
         self.cores = cores
-        if self.classes is not None:
+        if self.classes:
```

Comment:
if classes is an array a for loop over an empty array does nothing so you don't really need to wrap it in an if statement

Reply:
Setting the number of cores should be inside the if, actually...

Reply:
ah... another nice trick is to not use `else raise`

Reply:
```
        if not self.classes:
            raise Exception('You need load traffic classes before overwriting the number of cores')
      
        # continue as per normal
        self.cores = cores
        for c in self.classes:
            c.results.set_cores(cores)
                c._aon_results.set_cores(cores)
```

Reply:
Hahahahaha  I thought of the same thing while the tests ran locally... hahahaha